### ISSUE TITLE
Mistleading error message when 

### ISSUE THREAD

**Issue by bbarker** [purescript/purescript#3726]
This should be on 0.13.3. I've gotten a few errors like this while making typos while redefining a large collection of record types in terms of row types. Here's an example.

I get this error:

```
[1/1 KindsDoNotUnify] src/Metajelo/Types.purs:101:13

  101  , format :: Array Format
                   ^^^^^
  
  Could not match kind
  
    Type
  
  with kind
  
    # Type
  
  while checking the kind of ( basicMetadata :: BasicMetadata
                             , format :: Array Format
                             , location :: Location
                             , resourceID :: Maybe ResourceID
                             , resourceMetadataSource :: Maybe ResourceMetadataSource
                             , resourceType :: ResourceType
                             )
  in type synonym SupplementaryProductRows
```

From this code:

```purescript
type SupplementaryProductRows = (
  basicMetadata :: BasicMetadata
, resourceID :: Maybe ResourceID
, resourceType :: ResourceType
, format :: Array Format
, resourceMetadataSource :: Maybe ResourceMetadataSource
, location :: Location
)
type SupplementaryProduct = Record SupplementaryProductRows
```

Hmm, nothing odd there. But oh, I just redefined `ResourceType` also, which is the next line up from the `Array Format` error:

```purescript
type ResourceTypeRows = (
  description :: String
, generalType :: ResourceTypeGeneral
)
type ResourceType = ResourceTypeRows
```

Woops, forgot to put in a `Record` on the last line, should be:

```purescript
type ResourceType = Record ResourceTypeRows
```
That fixes it.

So in summary, the error isn't very helpful, and also points to the wrong (but succeeding) line.

**Comment by hdgarrood** [comment]
I think there are two things going on here. Firstly, the type error you get when you have kind mismatches in a row could be improved. For example:

```purescript
module Main where
type X = ( a :: Int, b :: Array )
```

currently gives:

```
Error found:
in module Main
at src/Main.purs:2:27 - 2:32 (line 2, column 27 - line 2, column 32)

  Could not match kind

    Type -> Type

  with kind

    Type


while checking the kind of ( a :: Int  
                           , b :: Array
                           )           
in type synonym X
```

But we could maybe make it clearer that the error is occurring because all of the kinds in the row need to match, perhaps with a supplemental hint. Perhaps it could look like this:

> Hint: All of the types appearing in a row must have the same kind. Here, Int has kind Type, whereas Array has kind Type -> Type.

The other thing is that this particular error does indeed seem to be in the wrong place. I think the error should be reported at the first label whose kind doesn't match the kinds of the labels before it, so in this case that ought to be `resourceType`. However this is hard to investigate without a self-contained example. Are you able to come up with one?

**Comment by Woody88** [comment]
Since the error context is `KindsDoNotUnify` and the line before displaying the row says `while checking the kind of` (the important word **kind**), why not just display the kinds of each label instead of their types and add the hint at the bottom? 

Something like so: 
```
Error found:
in module Main
at src/Main.purs:2:27 - 2:32 (line 2, column 27 - line 2, column 32)

  Could not match kind

    Type -> Type

  with kind

    Type


while checking the kind of ( a :: Type
                           , b :: Type -> Type
                           )           
in type synonym X

Hint: All of the types appearing in a row must have the same kind
```
Would it just further confuse people?

**Comment by hdgarrood** [comment]
Yeah, I don’t think that’s a great approach, as what you have there doesn’t really make sense; the thing whose kind was being checked there is not a type. For example, you can’t write `:kind ( a :: Type, b :: Type -> Type )` in the repl.
